rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+                .withRepository(repository)
+                .withEventLogger(mock(BridgeEventLogger.class))
+                .withExecutionBlock(mock(Block.class))
+                .build();
+
+        bridgeSupport.addSignature(federation.getBtcPublicKeys().get(0), null, createHash().getBytes());
+        bridgeSupport.save();
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(repository, PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstantsRegtest, activationsBeforeForks);
+
+        assertTrue(provider.getRskTxsWaitingForSignatures().isEmpty());
+    }
+
+    @Test
+    public void addSignatureFromInvalidFederator() throws Exception {
+
+        Repository repository = createRepository();
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstantsRegtest)
+                .withProvider(new BridgeStorageProvider(
+                        repository,
+                        PrecompiledContracts.BRIDGE_ADDR,
+                        bridgeConstantsRegtest,
+                        activationsBeforeForks))
+                .withRepository(repository)
+                .withEventLogger(mock(BridgeEventLogger.class))
+                .withExecutionBlock(mock(Block.class))
+                .build();
+
+        bridgeSupport.addSignature(new BtcECKey(), null, createHash().getBytes());
+        bridgeSupport.save();
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(repository, PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstantsRegtest, activationsBeforeForks);
+
+        assertTrue(provider.getRskTxsWaitingForSignatures().isEmpty());
+    }
+
+    @Test
+    public void addSignatureWithInvalidSignature() throws Exception {
+        addSignatureFromValidFederator(Lists.newArrayList(new BtcECKey()), 1, true, false, "InvalidParameters");
+    }
+
+    @Test
+    public void addSignatureWithLessSignaturesThanExpected() throws Exception {
+        List<BtcECKey> keys = Collections.singletonList(BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0));
+        addSignatureFromValidFederator(keys, 0, true, false, "InvalidParameters");
+    }
+
+    @Test
+    public void addSignatureWithMoreSignaturesThanExpected() throws Exception {
+        List<BtcECKey> keys = Collections.singletonList(BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0));
+        addSignatureFromValidFederator(keys, 2, true, false, "InvalidParameters");
+    }
+
+    @Test
+    public void addSignatureNonCanonicalSignature() throws Exception {
+        List<BtcECKey> keys = Collections.singletonList(BridgeRegTestConstants.REGTEST_FEDERATION_PRIVATE_KEYS.get(0));
+        addSignatureFromValidFederator(keys, 1, false, false, "InvalidParameters");
+    }
+
+    private void test_addSignature_EventEmitted(boolean rskip146Active, boolean rskip326Active, byte[] signature, int wantedNumberOfInvocations) throws Exception {
```

Comment:
`rskip146Active` is always true when you call it. And I think it makes sense, we just want to test how many times BridgeEventLogger was called not which implementation.

So I would just this parameter and set RSKIP146 to active when creating the activations mock